http/server_sent_event_message_stream.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#fields}
+ */
+export interface ServerSentEventMessage {
+  /** Ignored by the client. */
+  comment?: string;
+  /** A string identifying the type of event described. */
+  event?: string;
+  /** The data field for the message. Split by new lines. */
+  data?: string;
+  /** The event ID to set the {@linkcode EventSource} object's last event ID value. */
+  id?: string | number;
+  /** The reconnection time. */
+  retry?: number;
+}
+
+/**
+ * Converts a server-sent message object into a string for the client.
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format}
+ */
+function stringify(message: ServerSentEventMessage) {
+  const lines = [];
+  if (message.comment) lines.push(`:${message.comment}`);
+  if (message.event) lines.push(`event:${message.event}`);
+  if (message.data) {
+    message.data.split(/\r?\n/).forEach((line) => lines.push(`data:${line}`));
```

Comment:
I think just \r is also a valid new line delimiter. Please check the spec tho

Reply:
Yep, this line splits `messsage.data` by `\r\n`, `\n` and `\r`. In case there's any confusion, this is different to how newlines are represented in the SSE final message string, which is safe to just be a `\n`, according to the spec (see [here](https://html.spec.whatwg.org/multipage/server-sent-events.html#parsing-an-event-stream)). Either way, I've added to the test to check that `message.data` is split correctly for various delimiters.